packages/flame/lib/src/components/component.dart

```diff
@@ -319,6 +319,22 @@ class Component {
     }
   }
 
+  /// An iterator that returns all added components.
+  Iterable<Component> descendants({bool includeSelf = false}) sync* {
```

Comment:
I was thinking, for convenience only, having it generic as:
```
 Iterable<T> descendants<T extends Component>({bool includeSelf = false}) sync* 
```
But one could use `whereType`, although this is more verbose.

@st-pasha @Dan-Crane wdyt?

Reply:
Better to have it consistent with `ancestors`

Reply:
You're right

Reply:
@st-pasha in the old version of `ancestors()` it was:
```dart
Iterable<Component> ancestors() sync* {
  var current = parent;
  while (current != null) {
    yield current;
    current = current.parent;
  }
}
```
Should I stick to it or stick to new version with generic type?

Reply:
I prefer the non-template version, because there are already simple and well-known primitives to filter an iterable based on type. So, `ancestors().whereType<T>()` would work perfectly well instead of the templated version. On the other hand, if filtering by type is not needed, then there is no penalty for doing unnecessary work.

Reply:
leave the current implementation for `descendants()`? Also i can remake the ancestor method in this PR and write tests for it

Reply:
Agree with @alestiago